File: exo2-2.py
import os
import sys

pid_fils1 = os.fork()
if pid_fils1 != 1:
    if pid_fils1 == 0:
        print("Je suis le fils1")
    else:
        print("Je suis le père")
        pid_fils2 = os.fork()
        if pid_fils2 != -1:
            print("Je suis le fils2" if pid_fils2 == 0 else "Je suis le père")
        else:
            print("Erreur lors de la création de fils2")
else:
    print("Erreur lors de la création de fils1")

File: exo2-2.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>
#include <errno.h>

pid_t pid_fils1;
pid_t pid_fils2;

pid_t suivant(pid_t fils2)
{
    switch (pid_fils2=fork())
    {
        case -1 :
            perror("Problème dans fork() pour fils 2\n");
            exit(errno);
            break;
        
        case 0 :
            puts("Je suis le fils 2");
            exit(6);
        
        default :
            puts("Je suis le père");
            wait(NULL); 
            break;
    }
    return fils2;
}
int main(void)
{
    switch (pid_fils1=fork())
    {
        case -1 :
            perror("Problème dans fork() pour fils 1\n");
            exit(errno);
            break;
        
        case 0 :
            puts("Je suis le fils 1");
            exit(5);
        
        default : 
            suivant(pid_fils2);
            wait(NULL); 
            break;
    }
    return 0;
}